src/osemgrep/networking/Scan_helper.ml

```diff
@@ -52,20 +65,55 @@ let fetch_scan_config ~token ~sca ~dry_run ~full_scan repository =
   let content =
     let headers = [ ("authorization", "Bearer " ^ token) ] in
     match Http_helpers.get ~headers url with
-    | Ok body -> body
+    | Ok _ as r -> r
     | Error msg ->
-        (* was raise Semgrep_error, but equivalent to abort now *)
-        Error.abort
+        Error
           (Printf.sprintf "Failed to download config from %s: %s"
              (Uri.to_string url) msg)
   in
   Logs.debug (fun m -> m "finished downloading from %s" (Uri.to_string url));
-  try
-    match Yojson.Basic.from_string content with
-    | `Assoc e -> (
-        match List.assoc "rule_config" e with
-        | `String e -> e
-        | _else -> invalid_arg "couldn't retrieve config")
-    | _else -> invalid_arg "couldn't retrieve config"
-  with
-  | _failure -> invalid_arg "couldn't retrieve config"
+  match content with
+  | Error _ as e -> e
+  | Ok content -> (
+      try
+        match Yojson.Basic.from_string content with
```

Comment:
same, might be useful to have a separate xxx_of_json here that does this json parsing.

Reply:
done in [93bafdf](https://github.com/returntocorp/semgrep/pull/8149/commits/93bafdfd8882502b01454b97d93e2009c352ef8d)